packages/cli/src/oclif/commands/logout.js

```diff
@@ -0,0 +1,36 @@
+const BaseCommand = require('../ZapierBaseCommand');
+const { buildFlags } = require('../buildFlags');
+
+const { callAPI } = require('../../utils/api');
+const { deleteFile } = require('../../utils/files');
+const { AUTH_LOCATION, AUTH_LOCATION_RAW } = require('../../constants');
+
+class LogoutCommand extends BaseCommand {
+  async perform() {
+    this.log(
+      'Preparing to deactivate local deploy key and reset local configs.'
+    );
+    this.startSpinner('Deactivating local deploy key');
+    try {
+      await callAPI('/keys', { method: 'DELETE', body: { single: true } });
+    } catch (e) {
+      this.warn(
+        'Deletion API request failed. If this is unexpected, rerun this command with `--debug` for more info.'
+      );
```

Comment:
When the server fails to delete the key and returns an error, should we just exit? I know this follows the original implementation, but maybe it makes more sense not to continue if an error happens.

Another issue I found is when I tested `zapier logout` in a directory that doesn't have `.zapierapprc`, the output formatting is off:
![](https://zappy.zapier.com/1B475D1A-84B2-4516-90FD-6ECC045CF3C8.png)

I think adding a `this.endSpinner()` before printing the warning message should fix the formatting issue.

Reply:
used a `finally`! good catch.